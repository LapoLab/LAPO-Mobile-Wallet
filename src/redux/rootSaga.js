import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
