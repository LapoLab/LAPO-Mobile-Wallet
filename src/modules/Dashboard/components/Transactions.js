/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text, View, TouchableOpacity, FlatList,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';
import moment from 'moment';
import { bindActionCreators } from 'redux';
import {
  setModalTxDetails as setModalTxDetailsAction,
  setShowSpinner as setShowSpinnerAction,
} from '../../../commons/actions';
import calcAmount from '../../../utils/calcAmount';
import styles from './styles';

type Props = {
  navigation: {
    openDrawer: Function,
    dispatch: Function,
    navigate: Function,
  },

  transactions: [Object],
  setModalTxDetails: Function,
};
type State = {
  nameFilter: string,
};
class Dashboard extends Component<Props, State> {
  state = {
    nameFilter: 'all',
  };

  handleShowMenu = () => {
    const {
      navigation: { openDrawer },
    } = this.props;
    openDrawer();
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleSetFilter = nameFilter => () => {
    this.setState({ nameFilter });
  };

  handleChooseTx = (txId: string) => () => {
    const { setModalTxDetails } = this.props;
    setModalTxDetails(txId);
  };

  keyExtractor = item => item.id;

  renderTransactions = () => {
    const { transactions } = this.props;
    const { nameFilter } = this.state;
    const tranFilter = transactions.filter((item) => {
      if (nameFilter === 'outcome') {
        return item.mode === 'input';
      }
      if (nameFilter === 'income') {
        return item.mode === 'out';
      }
      return true;
    });

    const trans = tranFilter.sort(
      (a, b) => Number(b.time === 0 ? b.time2 : b.time) - Number(a.time === 0 ? a.time2 : a.time),
    );

    if (trans.length === 0) {
      return (
        <View style={styles.rowInOutEmpty}>
          <Text style={styles.textEmpty}>no transactions yet</Text>
        </View>
      );
    }

    return (
      <View style={{ paddingBottom: 50 }}>
        <FlatList
          data={trans}
          contentContainerStyle={{ paddingBottom: 0 }}
          keyExtractor={this.keyExtractor}
          renderItem={({ item: tran }) => (
            <TouchableOpacity
              style={styles.rowInOut}
              key={`${tran.id}`}
              onPress={this.handleChooseTx(tran.id)}
            >
              <View style={styles.RowInOut}>
                <Text style={tran.mode === 'out' ? styles.titleIn : styles.titleOut}>
                  {tran.time > 154262948 ? moment.unix(tran.time).format('DD MMM YYYY') : '-'}
                </Text>
                <Text style={tran.mode === 'out' ? styles.titleIn : styles.titleOut}>
                  {tran.mode === 'out' ? '+ ' : '- '}
                  {calcAmount.calcAmountLux(tran.amount)}
                  {' LAX'}
                </Text>
              </View>
              <View style={styles.RowInOut}>
                <Text style={styles.subTitleInOutLeft}>
                  {tran.time > 154262948 ? moment.unix(tran.time).format('HH:mm') : '-'}
                </Text>
                <Text style={styles.subTitleInOutRight}>{tran.address}</Text>
              </View>
            </TouchableOpacity>
          )}
        />
      </View>
    );
  };

  render() {
    const { nameFilter } = this.state;

    return (
      <View style={styles.transactionContainer}>
        <View style={styles.groupInOut}>
          <View style={styles.headerInOut}>
            <TouchableOpacity onPress={this.handleSetFilter('all')}>
              <Text style={nameFilter === 'all' ? styles.textInOutActive : styles.textInOut}>
                ALL
              </Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={this.handleSetFilter('income')}>
              <Text style={nameFilter === 'income' ? styles.textInOutActive : styles.textInOut}>
                INCOME
              </Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={this.handleSetFilter('outcome')}>
              <Text style={nameFilter === 'outcome' ? styles.textInOutActive : styles.textInOut}>
                OUTCOME
              </Text>
            </TouchableOpacity>
          </View>

          <View style={styles.bodyInOut}>{this.renderTransactions()}</View>
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => ({
  address: state.wallet.address,
  transactions: state.wallet.transactions,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setModalTxDetails: setModalTxDetailsAction,
    setShowSpinner: setShowSpinnerAction,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Dashboard);
