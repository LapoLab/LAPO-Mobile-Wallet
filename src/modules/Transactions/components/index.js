/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  Image,
  SafeAreaView,
  BackHandler,
  FlatList,
  Platform,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';
import moment from 'moment';
import { bindActionCreators } from 'redux';

import SpinnerPage from '../../../commons/components/SpinnerPage';
import Modals from '../../Dashboard/components/Modals';
import {
  setModalTxDetails as setModalTxDetailsAction,
  setShowSpinner as setShowSpinnerAction,
} from '../../../commons/actions';
import calcAmount from '../../../utils/calcAmount';
import * as actionNotify from '../../Notify/actions';
import Notify from '../../Notify/component';
import styles from './styles';
import {
  menuIcon, ActivityIcon, iconNotifyActive, dashboardIcon, triangle,
} from '../../../assets';

type Props = {
  navigation: { openDrawer: Function, dispatch: Function, navigate: Function },
  transactions: [Object],
  setVisibleNotify: Function,
  isVisibleNotify: boolean,
  isOpenDrawer: boolean,
  timeNotifyLast: number,
  txDetails: string,
  setModalTxDetails: Function,
  notes: Object,
};
type State = {
  nameFilter: string,
};

class Transactions extends Component<Props, State> {
  state = {
    nameFilter: 'all',
  };

  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBack);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBack);
  }

  handleBack = () => {
    const { navigation: { openDrawer }, isOpenDrawer } = this.props;
    if (isOpenDrawer) {
      this.navigateToScreen('Dashboard')();
      return true;
    }
    openDrawer();
    return true;
  };

  handleShowMenu = () => {
    const {
      navigation: { openDrawer },
    } = this.props;
    openDrawer();
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleSetFilter = nameFilter => () => {
    this.setState({ nameFilter });
  };

  handleShowNotify = () => {
    const { setVisibleNotify, isVisibleNotify } = this.props;
    setVisibleNotify(!isVisibleNotify);
  };

  handleCLoseModal = () => {
    const { setModalTxDetails } = this.props;
    setModalTxDetails('');
  };

  handleChooseTx = (txId: string) => () => {
    const { setModalTxDetails } = this.props;
    setModalTxDetails(txId);
  };

  renderTiangle = () => {
    const { nameFilter } = this.state;
    return (
      <View style={styles.conrainerTriangle}>
        <View style={styles.activeIconColumn}>
          {nameFilter === 'all' && <Image style={styles.iconActiveInOut} source={triangle} />}
        </View>
        <View style={styles.activeIconColumn2}>
          {nameFilter === 'income' && <Image style={styles.iconActiveInOut} source={triangle} />}
        </View>
        <View style={styles.activeIconColumn}>
          {nameFilter === 'outcome' && <Image style={styles.iconActiveInOut} source={triangle} />}
        </View>
      </View>
    );
  };

  keyExtractor = item => item.id;

  renderNote = (tran) => {
    const { notes } = this.props;
    if (!notes[tran.id]) return null;
    if (notes[tran.id] && notes[tran.id] === '') return null;

    return (
      <View style={styles.RowInOutForNote}>
        <Text style={styles.labelNote}>Note:</Text>
        <Text style={styles.valueNote}>{notes[tran.id] ? `${notes[tran.id]}` : '-'}</Text>
      </View>
    );
  };

  renderTransactions = () => {
    const { transactions } = this.props;
    const { nameFilter } = this.state;
    const tranFilter = transactions.filter((item) => {
      if (nameFilter === 'outcome') {
        return item.mode === 'input';
      }
      if (nameFilter === 'income') {
        return item.mode === 'out';
      }
      return true;
    });

    const trans = tranFilter.sort(
      (a, b) => Number(b.time === 0 ? b.time2 : b.time) - Number(a.time === 0 ? a.time2 : a.time),
    );

    if (trans.length === 0) {
      return (
        <View style={styles.rowInOutEmpty}>
          <Text style={styles.textEmpty}>no transactions yet</Text>
        </View>
      );
    }

    return (
      <View style={{ marginBottom: 0 }}>
        <FlatList
          data={trans}
          keyExtractor={this.keyExtractor}
          renderItem={({ item: tran }) => (
            <TouchableOpacity
              style={styles.rowInOut}
              key={`${tran.id}`}
              onPress={this.handleChooseTx(tran.id)}
            >
              <View style={styles.RowInOut}>
                <Text style={tran.mode === 'out' ? styles.titleIn : styles.titleOut}>
                  {tran.time > 154262948 ? moment.unix(tran.time).format('DD MMM YYYY') : '-'}
                </Text>
                <Text style={tran.mode === 'out' ? styles.titleIn : styles.titleOut}>
                  {tran.mode === 'out' ? '+ ' : '- '}
                  {calcAmount.calcAmountLux(tran.amount) || ''}
                  {' LAX'}
                </Text>
              </View>
              <View style={styles.RowInOut}>
                <Text style={styles.subTitleInOutLeft}>
                  {tran.time > 154262948 ? moment.unix(tran.time).format('HH:mm') : '-'}
                </Text>
                <Text style={styles.subTitleInOutRight}>{tran.address || ' '}</Text>
              </View>
              {this.renderNote(tran)}
            </TouchableOpacity>
          )}
        />
      </View>
    );
  };

  render() {
    const { nameFilter } = this.state;
    const {
      isVisibleNotify, transactions, timeNotifyLast, txDetails, navigation,
    } = this.props;

    const modalVisible = txDetails !== '';

    const txsWithoutOut = transactions.filter(item => item.mode === 'out');

    const isNotifyActive = !!(
      txsWithoutOut.length > 0 && txsWithoutOut[txsWithoutOut.length - 1].time > timeNotifyLast
    );

    return (
      <LinearGradient
        start={{ x: -0.1, y: -0.1 }}
        end={{ x: 1.2, y: 1.2 }}
        locations={[0, 1]}
        colors={['#4183BD', '#093B0C']}
        style={{ flex: 1 }}
      >
        <SafeAreaView style={styles.container}>
          <View style={styles.header}>
            <TouchableOpacity style={styles.buttonMenu} onPress={this.handleShowMenu}>
              <Image style={styles.menuHeader} source={menuIcon} />
            </TouchableOpacity>

            <Image source={dashboardIcon} style={styles.dashboardIcon} />

            <TouchableOpacity
              style={styles.buttonNotify}
              onPress={isNotifyActive ? this.handleShowNotify : null}
            >
              <Image
                style={styles.notifyHeader}
                source={isNotifyActive ? iconNotifyActive : ActivityIcon}
              />
            </TouchableOpacity>
          </View>

          <View style={styles.body}>
            <View style={styles.transactionContainer}>
              <View style={styles.groupInOut}>
                <View style={styles.headerInOut}>
                  <TouchableOpacity onPress={this.handleSetFilter('all')}>
                    <Text style={nameFilter === 'all' ? styles.textInOutActive : styles.textInOut}>
                      ALL
                    </Text>
                  </TouchableOpacity>
                  <TouchableOpacity onPress={this.handleSetFilter('income')}>
                    <Text
                      style={nameFilter === 'income' ? styles.textInOutActive : styles.textInOut}
                    >
                      INCOME
                    </Text>
                  </TouchableOpacity>
                  <TouchableOpacity onPress={this.handleSetFilter('outcome')}>
                    <Text
                      style={nameFilter === 'outcome' ? styles.textInOutActive : styles.textInOut}
                    >
                      OUTCOME
                    </Text>
                  </TouchableOpacity>
                </View>

                <View style={styles.bodyInOut}>{this.renderTransactions()}</View>
              </View>
            </View>
          </View>
        </SafeAreaView>
        <View style={styles.bottomPlaceForX} />

        {Platform.OS !== 'ios' && modalVisible && <View style={styles.blurBackground} />}

        {isVisibleNotify && isNotifyActive && <Notify />}

        <SpinnerPage />

        {Platform.OS === 'ios' && <Modals navigation={navigation} />}

      </LinearGradient>
    );
  }
}

const mapStateToProps = state => ({
  transactions: state.wallet.transactions,
  isVisibleNotify: state.notify.isVisibleNotify,
  timeNotifyLast: state.notify.timeLast,
  txDetails: state.modals.txDetails,
  notes: state.wallet.notes,
  isOpenDrawer: state.modals.isOpenDrawer,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setVisibleNotify: actionNotify.setVisibleNotify,
    setModalTxDetails: setModalTxDetailsAction,
    setShowSpinner: setShowSpinnerAction,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Transactions);
