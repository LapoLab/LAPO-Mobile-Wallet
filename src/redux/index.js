import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
