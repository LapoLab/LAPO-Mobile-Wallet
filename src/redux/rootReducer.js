import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
