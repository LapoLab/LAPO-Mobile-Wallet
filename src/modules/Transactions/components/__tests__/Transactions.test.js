// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Transactions from '../index';

describe('Transactions page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Transactions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
