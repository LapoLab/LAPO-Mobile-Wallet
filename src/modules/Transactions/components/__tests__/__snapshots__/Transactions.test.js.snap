// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
        "zIndex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "elevation": 6,
          "flexDirection": "row",
          "height": 120,
          "justifyContent": "space-between",
          "marginTop": 30,
          "paddingHorizontal": 10,
          "shadowColor": "rgb(0,0,0)",
          "shadowOffset": Object {
            "height": 4,
            "width": 2,
          },
          "shadowOpacity": 0.34,
          "zIndex": 0,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 18,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/menuIcon.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "marginLeft": 8,
              "resizeMode": "contain",
              "width": 18,
            }
          }
        />
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/logo-dashboard.png",
          }
        }
        style={
          Object {
            "height": 100,
            "resizeMode": "contain",
            "width": 100,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 18,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/Activity.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "marginRight": 8,
              "resizeMode": "contain",
              "width": 18,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#EFF4FC",
          "flex": 4,
          "paddingBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 15,
              "elevation": 10,
              "flex": 1,
              "marginBottom": 15,
              "marginHorizontal": 30,
              "marginTop": 25,
              "padding": 2,
              "shadowColor": "rgb(0,0,0)",
              "shadowOffset": Object {
                "height": 4,
                "width": 2,
              },
              "shadowOpacity": 0.34,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4183BD",
                "borderTopLeftRadius": 15,
                "borderTopRightRadius": 15,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 18,
                  }
                }
              >
                ALL
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Light",
                    "fontSize": 18,
                    "opacity": 0.7,
                  }
                }
              >
                INCOME
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Light",
                    "fontSize": 18,
                    "opacity": 0.7,
                  }
                }
              >
                OUTCOME
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomLeftRadius": 15,
                "borderBottomRightRadius": 15,
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Lato-Light",
                    "marginBottom": 12,
                  }
                }
              >
                no transactions yet
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "backgroundColor": "#EFF4FC",
        "height": 0,
      }
    }
  />
</BVLinearGradient>
`;
